src/components/layout/RegisterNameForm/RegisterNameForm.tsx

```diff
@@ -0,0 +1,159 @@
+        <div className="register-inputs center">
+          <div className="input-group center column">
+            <Dropdown
+              showSelected={true}
+              showChevron={true}
+              selected={antFlow}
+              setSelected={setAntFlow}
+              options={{
+                'Create an Arweave Name Token (ANT) for me':
+                  'Create an Arweave Name Token (ANT) for me',
+                'I have an Arweave Name Token (ANT) I want to use':
+                  'I have an Arweave Name Token (ANT) I want to use',
+              }}
+            />
+            {/**TODO: add global error state to mark issues with the target id */}
+            {antFlow === 'Create an Arweave Name Token (ANT) for me' ? (
+              <input
+                className="data-input center"
+                placeholder="Target ID"
+                onChange={(e) => handleTargetID(e.target.value)}
+                maxLength={43}
+                style={
+                  targetID && isArweaveTransactionID(targetID)
+                    ? { border: '4px solid var(--success-green)' }
+                    : targetID && isArweaveTransactionID(targetID)
+                    ? { border: '4px solid var(--error-red)' }
+                    : {}
+                }
+              />
+            ) : antFlow ===
+              'I have an Arweave Name Token (ANT) I want to use' ? (
+              <Dropdown
+                showChevron={true}
+                showSelected={false}
+                options={['ant one', 'ant two', 'ant three', 'ant four']}
+                selected={antAddress}
+                setSelected={setAntAddress}
+                headerElement={
+                  <SearchBar
+                    validationPredicate={(name) =>
+                      isArNSDomainNameValid({ name })
+                    }
+                    successPredicate={() => {
+                      return true;
+                    }}
+                    placeholderText="Enter ANT Contract ID"
+                    height={30}
+                  />
+                }
+                footerElement={
+                  <div
+                    className="data-input last-dropdown-option center"
+                    style={{ background: 'var(--bg-color)' }}
+                  >
+                    <span className="text bold white center">
+                      See All Tokens
+                    </span>
+                  </div>
+                }
+              />
+            ) : (
+              <></>
+            )}
+          </div>
+          {/**add a tag component for multiple controllers seperated by "," commas. In the future might have an address book. Possibly suggest other accounts from arconnect. */}
+          <input
+            className="data-input center"
+            type="text"
+            placeholder="Controllers"
+            onChange={(e) => handleTargetID(e.target.value)}
+            maxLength={43}
```

Comment:
disallow whitespace and special chars (except `-` and `_`) for all of these inputs

Reply:
fixed